content/01_WhyRx.md

```diff
@@ -1,241 +1,126 @@
+
+(A brief grammatical aside: although the phrase "Reactive Extensions" is plural, when we reduce it to just Rx.NET or Rx, we treat it as a singular noun. This is inconsistent, but saying "Rx are..." sounds plain weird.)
+
+For example, C# has integrated query features that we might use to find all of the entries in a list that meet some criteria. If we have some `List<Trade> trades` variable, we might write this:
+
+```cs
+var bigTrades =
+    from trade in trades
+    where trade.Volume > 1_000_000;
+```
+
+With Rx, we could use this exact same code with live data. Instead of being a `List<Trade>`, the `trades` variable could be an `IObservable<Trade>`. (`IObservable<T>` is the fundamental abstraction in Rx. It is essentially a live version of `IEnumerable<T>`.) In this case, `bigTrades` would also be an `IObservable<Trade>`, a live data source able to notify us of all trades whose `Volume` exceeds one million. Crucially, it will report each such trade immediately—this is what we mean by a 'live' data source.
+
+Rx is a powerfully productive development tool. It enables developers to work with live event streams using language features familiar to all .NET developers. It enables an elegant, declarative approach that often allows us to express complex behaviour more elegantly and with less code than would be possible without Rx.
+
+Rx builds on LINQ (Language Integrated Query). This enables us to use the query syntax shown above (or you can use the explicit function call approach that some .NET developers prefer). LINQ is widely used in .NET both for data access (e.g., in Entity Framework Core), but also for working with in-memory collections (with LINQ to Objects), meaning that experienced .NET developers will tend to feel at home with Rx. Crucially, LINQ is a highly composable design: you can connect operators together in any combination you like, expressing potentially complex processing in a straightforward way.
+
+
+## When is Rx appropriate?
+
+Rx is designed for processing sequences of events, meaning that it suits some scenarios better than others. The next sections describe some of these scenarios, and also cases in which it is a less obvious match but still worth considering. Finally, we describe some cases in which it is possible to use Rx but where alternatives are likely to be better.
+
+
+### Good Fit with Rx
+
+Rx is well suited to representing events that originate from outside of your code, and which your application needs to respond to, such as:
+
+- Integration events like a broadcast from a message bus or a push event from WebSockets API or other low latency middleware like [Azure Event Grid](https://azure.microsoft.com/en-gb/products/event-grid/), [Azure Event Hubs](https://azure.microsoft.com/en-gb/products/event-hubs/) and [Azure Service Bus](https://azure.microsoft.com/en-gb/products/service-bus/)
```

Comment:
I'm wondering if we could work a cloud standard like https://cloudevents.io/ into either this section, or a demo in the book...